### PR TITLE
Add Jenkins parameterized build config file

### PR DESCRIPTION
**Pull request by oskarth**
image

Addresses https://github.com/status-im/status-react/issues/2511

reviewers: sanity check (actual code is in Jenkins anyway)

testers: feedback on what you want to toggle, and test parameterized build

status: ready

### CODE REVIEW

File: Jenkinsfile.parameters [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            ipaUrl = 'https://i.diawi.com/' + hash
+
+            sh ('echo ARTIFACT iOS: ' + ipaUrl)
+
+        }
+    }
+
+    stage('Slack Notification') {
+            def c = (testPassed ? 'good' : 'warning' )
+            slackSend color: c, message: 'Branch: ' + BRANCH_NAME +
```
Comment (oskarth) [inline Jenkinsfile.parameters]: Should probably print repo

**Comment by oskarth** [comment]
Moved to 'ready for testing', possibly prematurely for two  reasons:

1) Code that is live is actually live already at Jenkins by necessity - only way to test, ephemeral config and I kind of overwrote it :morty:

2) Want to make sure normal external contributor PRs still work, hence testing stage, this is just a copy of this.

Regardless it'd be good to get a review on this, pinging 9-dev-xp swarm for thoughts. See comments for known downsides https://github.com/status-im/status-react/issues/2511#issuecomment-348318532 - if you have solutions shoot. I know it can be shell-fued but didn't have time to make it robust enough, hence this simplistic approach.

**Comment by annadanchenko** [comment]
@rasom what parameters do we toggle for the production release and with what values? It would be great if we can just build the release candidates on our own using the params in Jenkins. So, if some of the "production" parameters is missing, please tell which one. And of course which values to use :)

**Comment by oskarth** [comment]
@annadanchenko https://github.com/status-im/status-react/blob/develop/.env.prod

Additionally some certificate stuff, but we can defo do RC this way :)

**Comment by oskarth** [comment]
Before merge: probably fix this https://github.com/status-im/status-react/pull/2553#pullrequestreview-81289251

**Comment by rasom** [comment]
@oskarth i also added repo name to slack messages in jenkins configurations, it makes a bit simpler to distinguish parametrized builds from these which are built form branches, like
```groovy
slackSend color: 'good', message: REPO + ":" + BRANCH_NAME + ' build started. ' + env.BUILD_URL
```

and 

```groovy
slackSend color: c, message: 'Branch: ' + REPO + ":" + BRANCH_NAME +
```

File: Jenkinsfile.parameters [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      sh 'echo MAINNET_NETWORKS_ENABLED='  + MAINNET_NETWORKS_ENABLED  + '>>' + '.env'
+      sh 'echo ERC20_ENABLED='             + ERC20_ENABLED             + '>>' + '.env'
+      sh 'echo LOG_LEVEL='                 + LOG_LEVEL                 + '>>' + '.env'
+      sh 'echo "**********************************************************************"'
+      sh 'echo PARAMETERIZED BUILD - USING CUSTOM ENVIRONMENT'
+      sh 'cat .env'
+      sh 'echo "**********************************************************************"'
+
+      sh 'lein deps && npm install && ./re-natal deps'
+      sh 'sed -i "" "s/301000/1201000/g" node_modules/react-native/packager/src/JSTransformer/index.js'
```
Comment (yenda) [inline Jenkinsfile.parameters]: I think we should remove this line now, also once react-native will be updated this won't work anymore as the packager is now a separate project

Comment (oskarth) [reply Jenkinsfile.parameters]: @yenda Can you raise an issue and we test/do this in a separate PR and for all Jenkinsfiles? This is not in scope of this PR IMO (and also need to update it in Jenkins).

Comment (rasom) [reply Jenkinsfile.parameters]: let's just try it on jenkins first

**Comment by oskarth** [comment]
Thanks @rasom, I'll look at this file in Jenkins and put it in this file, then merge. Soonish.

File: Jenkinsfile.parameters (hunk 2 above)
Comment (yenda) [reply Jenkinsfile.parameters]: It was already commented out in jenkins until @oskarth updated it with that script

**Comment by oskarth** [comment]
FYI, added this to live version (with same default values as env.jenkins):
```
sh 'echo JSC_ENABLED='               + JSC_ENABLED               + '>>' + '.env'
sh 'echo OFFLINE_INBOX_ENABLED='     + OFFLINE_INBOX_ENABLED     + '>>' + '.env'
```

Also uncommented back `sh 'sed -i "" "s/301000/1201000/g" node_modules/react-native/packager/src/JSTransformer/index.js'` since all parameterized builds are failing without it right now.

**Comment by yenda** [comment]
@oskarth No, it was update-react-native build only that was failing and it is not related to this. Now it will fail because of the sed

**Comment by oskarth** [comment]
@rasom @yenda updated to reflect current Jenkins config at jenkins.status.im - if you have no objections, can you please approve and lets merge this?

**Review by rasom — APPROVED** [review]

**Pull request merged**